### PR TITLE
[4/x] Updated transactional tests

### PR DESCRIPTION
**Pull request by tnowacki**
- Updated transactional tests to remove dynamic child usage
- No new tests

- Built on #5169

### CODE REVIEW

File: crates/sui-adapter-transactional-tests/tests/child_count/count_decremented.move [hunk 1]
```diff
@@ -10,138 +10,91 @@
 
 module test::m {
     use sui::tx_context::{Self, TxContext};
+    use sui::dynamic_object_field as dof;
```
Comment (amnn) [inline crates/sui-adapter-transactional-tests/tests/child_count/count_decremented.move:13]: In the spirit of making this an example that others can copy-pasta, `dof` seems a bit too opaque at the call-site, how about `field` or `ofield`?

File: crates/sui-adapter-transactional-tests/tests/child_count/delete_by_wrap.move [hunk 2]
```diff
@@ -10,31 +10,45 @@
...
-        sui::transfer::transfer_to_object(child, parent)
+    public entry fun remove_and_add(parent: &mut S, idx: u64) {
+        let child: S = dof::remove(&mut parent.id, idx);
+        dof::add(&mut parent.id, idx, child)
+    }
+
+    public entry fun remove_and_wrap(parent: &mut S, idx: u64, ctx: &mut TxContext) {
+        let child: S = dof::remove(&mut parent.id, idx);
+        dof::add(&mut parent.id, idx, R { id: sui::object::new(ctx), s: child })
+    }
```
Comment (amnn) [inline crates/sui-adapter-transactional-tests/tests/child_count/delete_by_wrap.move:47]: Are these necessary?

File: crates/sui-adapter-transactional-tests/tests/child_count/delete_parent_invalid.move [hunk 3]
```diff
@@ -10,36 +10,56 @@
...
-        sui::transfer::transfer_to_object(child, parent)
+    public entry fun remove_and_add(parent: &mut S, idx: u64) {
+        let child: S = dof::remove(&mut parent.id, idx);
+        dof::add(&mut parent.id, idx, child)
+    }
+
+    public entry fun remove_and_wrap(parent: &mut S, idx: u64, ctx: &mut TxContext) {
+        let child: S = dof::remove(&mut parent.id, idx);
+        dof::add(&mut parent.id, idx, R { id: sui::object::new(ctx), s: child })
     }
```
Comment (amnn) [inline crates/sui-adapter-transactional-tests/tests/child_count/delete_parent_invalid.move:47]: Ditto here, it looks like we don't use these.

File: crates/sui-adapter-transactional-tests/tests/child_count/delete_parent_invalid.move [hunk 4]
```diff
@@ -10,36 +10,56 @@
...
 
     public entry fun delete(s: S) {
         let S { id } = s;
         sui::object::delete(id)
     }
 
+    public entry fun wrap(s: S, ctx: &mut TxContext) {
+        let r = R { id: sui::object::new(ctx), s };
+        sui::transfer::transfer(r, tx_context::sender(ctx))
+    }
```
Comment (amnn) [inline crates/sui-adapter-transactional-tests/tests/child_count/delete_parent_invalid.move:57]: Or this?

**Review by amnn — COMMENTED** [review]
My favourite part of reviewing this PR was realising I'd already reviewed most of it in #5169!  I focussed on the transactional tests, which I think look fine?  I noticed what I thought were a lot of unused definitions, but there were quite a few which makes me think I don't really understand that testing framework, so holding off on the accept until my suspicions can be confirmed or denied.

File: crates/sui-adapter-transactional-tests/tests/child_count/count_decremented.move (hunk 1 above)
Comment (tnowacki) [reply crates/sui-adapter-transactional-tests/tests/child_count/count_decremented.move:13]: `ofield` isn't bad. I use `dof` in a few spots.. so let me switch them to `ofield`

**Comment by tnowacki** [comment]
> I noticed what I thought were a lot of unused definitions, but there were quite a few which makes me think I don't really understand that testing framework, so holding off on the accept until my suspicions can be confirmed or denied.

It's just a result of me copy-pasting the same module over multiple tests. This makes it a lot easier for me to quickly jump between tests, but yeah, not so good for a reviewer :)

**Review by amnn — APPROVED** [review]
LGTM!

I do have a slight preference of getting rid of the unused functions in each test, once they've been finalised (not a blocking issue).  The thing I want to avoid is having duplicates of a function that aren't being run in test now, where we run the risk of its behaviour needing to change, it getting updated in some places but not others, not noticing because they aren't getting run, and then later depending on the old broken logic in a test when we spot that we can re-use that now stale function.

If it's possible to factor out the helper modules in a place where they can be used by multiple transactional tests, that would address this concern as well!

**Pull request merged**